### DISCUSSION TITLE
Error while opening file in Quatro while rendering. Please help

### DISCUSSION THREAD

**Discussion by hypersonique** [quarto-dev/quarto-cli#6426] [Q&A] [answered]
### Description

Hi, I am new to Quatro, and have probably a very dumb question, but here we are. When rendering my first Quatro documnet I get a messagge: Error while opening file, and then this: 22% (unnamed-chunk-1)Quitting from lines 20-24 (DSS-Validation-Results-Quarto.rmarkdown)  Error in read_sav("onlydss.sav") : could not find function "read_sav" Calls: .main ... withVisible -> eval_with_user_handlers -> eval -> eval                                                                                                              Execution halted. Would appreciate any help I can get. Thank you so much!

**Comment by mcanouil** [comment, marked as answer]
Welcome to Quarto!

The issue is not related to Quarto, but to your R code:
> `Error in read_sav("onlydss.sav") : could not find function "read_sav"`

As for regular R script, you need to either load the R package from which you are using `read_sav()` or prefix the function with the name of the R package `pkg_name::read_sav()`.

Note that, not sharing actual code makes helping you difficult.
Sharing a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository can help you help yourself and can help others to help you (here and everywhere).

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by hypersonique** [reply]
**Last update**: I deleted the quatro files and tryed rendering again and it worked. Thank you very much for your help!

Hi,

Thanx for a quick answer. I just started so there is not much in my document, only these two chunks of code: 
#Import data
nor <- read_sav("onlydss.sav")
aus <- readxl::read_excel("final data set of DSS.xlsx")[, 2:51]
colnames(aus) <- str_remove_all(names(aus), "0")

#Missing data analysis
all <- readRDS("complete.data")
names(all)[11:60] <- names(nor)
naniar::vis_miss(all)

When you say load the package, you mean that I need to load the package in Quatro? Packages are loaded and working fine in RStudio, and since Quatro is integrated I just assumed that there is no need to load them. If that is the case, do I just add a code junk and load packages in the Quatro document? Thank you so much for the reply, I really appreaciate it!

**Update:**
Loading the packages did defintaely help :).
#load packages
library(haven)
library(readxl)
library(psych)
library(lavaan)
library(stringr)

But now, Ive gotten this error message:

processing file: DSS-Validation-Results-Quarto.rmarkdown
                                                                                                            
output file: DSS-Validation-Results-Quarto.knit.md

Error in file(con, "w") : cannot open the connection
Calls: .main ... extract_preserve_chunks -> <Anonymous> -> write_utf8 -> writeLines -> file
In addition: Warning message:
In file(con, "w") :
  cannot open file './DSS-Validation-Results-Quarto.knit.md': Invalid argument
Execution halted
